cdap-docs/reference-manual/source/http-restful-api/lifecycle.rst

```diff
@@ -256,87 +272,256 @@ The response will be the same JSON array with additional parameters for each of
+   * - ``<program-id>``
+     - Name of the *flow*, *MapReduce*, *schedule*, *Spark*, *workflow*, *worker*, or *custom service*
+       being called
 
-When starting an program, you can optionally specify runtime arguments as a JSON map in the request body::
+A program that is in the STOPPED state cannot be stopped. If there are multiple runs of the program
+in the RUNNING state, this call will stop one of the runs, but not all of the runs. 
 
-  POST <base-url>/namespaces/default/apps/HelloWorld/flows/WhoFlow/start
+For example::
 
-with the arguments as a JSON string in the body::
+  POST <base-url>/namespaces/default/apps/HelloWorld/flows/WhoFlow/stop
 
-  {"foo":"bar","this":"that"}
+will stop the *WhoFlow* flow in the *HelloWorld* application.
 
-CDAP will use these these runtime arguments only for this single invocation of the
-program.
 
-.. topic::  **Note: Runtime Arguments RESTful API Deprecated**
+.. _http-restful-api-lifecycle-stop-run:
 
-    As of *CDAP v2.8.0*, *Runtime Arguments RESTful API* have been deprecated, pending removal in a later version.
-    Replace all use of *Runtime Arguments RESTful API* with :ref:`Preferences RESTful API <http-restful-api-v3-preferences>`.
-    *Preferences RESTful API* will have feature-parity with *Runtime Arguments RESTful API* as of the version in which
-    *Runtime Arguments RESTful API* are removed.
+Stop a Program Run
+------------------
+You can stop a specific run of a program by submitting an HTTP POST request::
 
-To save the runtime arguments so that CDAP will use them every time you start the program,
-issue an HTTP PUT with the parameter ``runtimeargs``::
+  POST <base-url>/namespaces/<namespace>/apps/<app-id>/<program-type>/<program-id>/runs/<run-id>/stop
 
-  PUT <base-url>/namespaces/default/apps/HelloWorld/flows/WhoFlow/runtimeargs
+.. list-table::
+   :widths: 20 80
+   :header-rows: 1
 
-with the arguments as a JSON string in the body::
+   * - Parameter
+     - Description
+   * - ``<namespace>``
+     - Namespace ID
+   * - ``<app-id>``
+     - Name of the application being called
+   * - ``<program-type>``
+     - One of ``flows``, ``mapreduce``, ``schedules``, ``spark``, ``workflows``, ``workers``, or ``services``
+   * - ``<program-id>``
+     - Name of the *flow*, *MapReduce*, *schedule*, *Spark*, *workflow*, *worker*, or *custom service*
+       being called
+   * - ``<run-id>``
+     - Run id of the run
+
+For example::
+
+  POST <base-url>/namespaces/default/apps/PurchaseHistory/mapreduce/PurchaseHistoryBuilder/runs/631bc459-a9dd-4218-9ea0-d46fb1991f82/stop
+
+will stop the specific run of the *PurchaseHistoryBuilder* mapreduce in the *PurchaseHistory* application.
+
+.. _http-restful-api-lifecycle-stop-multiple:
+
+Stop multiple Programs
+----------------------
+You can stop multiple programs from different applications and program types
+by submitting an HTTP POST request::
```

Comment:
not as it is now.  If this changes (CDAP-4162), will be sure to update these docs.